app/services/catalog_remediator.rb

```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+
+# Remediates catalog entries, e.g., failed replication records
+class CatalogRemediator
+  def self.prune_replication_failures(druid:, version:)
+    new(druid: druid, version: version).prune_replication_failures
+  end
+
+  def initialize(druid:, version:)
+    @druid = druid
+    @version = version
+  end
+
+  def prune_replication_failures
+    zipped_moab_versions_with_errors.map do |zipped_moab_version, error_results|
+      zip_parts = ZipPart.where(zipped_moab_version: zipped_moab_version)
+      Rails.logger.info(
+        "Replication failure error(s) found with #{druid} (v#{version}): #{error_results}\n" \
+        "Destroying zip parts (#{zip_parts.pluck(:id)}) and zipped moab version (#{zipped_moab_version.id})"
+      )
+
+      ApplicationRecord.transaction do
+        # NOTE: The order here matters! ZipPart instances must be destroyed before the ZippedMoabVersion
+        zip_parts.destroy_all
+        zipped_moab_version.destroy
+      end
+
+      [zipped_moab_version.version, zipped_moab_version.zip_endpoint.endpoint_name]
+    end
+  end
+
+  private
+
+  attr_reader :druid, :version
+
+  delegate :check_child_zip_part_attributes, to: Audit::CatalogToArchive
+
+  def preserved_object
+    PreservedObject.find_by(druid: druid)
+  end
+
+  def zip_cache_expiry_timestamp
+    Settings.zip_cache_expiry_time.to_i.minutes.ago
+  end
+
+  def zipped_moab_versions_beyond_expiry
+    preserved_object.zipped_moab_versions.where(version: version, created_at: zip_cache_expiry_timestamp..)
+  end
+
+  def zipped_moab_versions_with_errors
+    zipped_moab_versions_beyond_expiry.to_a.filter_map do |zipped_moab_version|
+      results = AuditResults.new(druid, version, zipped_moab_version.zip_endpoint)
+      next unless check_child_zip_part_attributes(zipped_moab_version, results)
+
+      zipped_moab_version.zip_endpoint.audit_class.check_replicated_zipped_moab_version(zipped_moab_version, results)
+      # TODO: "set check_unreplicated_parts to true"
```

Comment:
This is a recommendation @jmartin-sul made on the ticket. I'll see to this soon.